generators/generators-commons/src/main/java/uk/gov/justice/services/generators/commons/helper/MessagingResourceUri.java

```diff
@@ -26,7 +26,16 @@ public String destinationName() {
      * @return capitalised uri string without delimiters
      */
     public String toCapitalisedString() {
-        return capitalize(stripStart(uriString, "/"), '.').replace(".", "");
+        //Remove any character that is invalid as starting character for a class name
+        String uriStringStriped = uriString.replaceAll("^[^A-Za-z_$]+", "");
+
+        final StringBuilder nonWordCharAsString = new StringBuilder();
+        uriString.chars().filter(e -> !Character.isLetter(e)).forEach(e -> nonWordCharAsString.append((char) e));
+        char[] delimiters = nonWordCharAsString.toString().toCharArray();
+        String capitalizedWithCharacter = capitalize(uriStringStriped, delimiters);
```

Comment:
That looks much better but need also add numbers to the delimiters otherwise it will not work.
so the second line need to be:
     captialised = capitaliseFully(cleaned, " _$0123456789".toCharArray());